index.js

```diff
@@ -4,16 +4,16 @@ const discovery = require('hyperdiscovery')
 
 module.exports = function (archiveOrKey, opts) {
   var archive = archiveOrKey.key ? archiveOrKey : createArchive(archiveOrKey)
```

Comment:
You can use something [like this](https://github.com/mafintosh/hypercore-archiver/blob/master/index.js#L206). But there may be a more straightforward implementation I can try but need to go get ☕️ first :).